java/core/src/main/java/com/microsoft/bond/Bonded.java

```diff
@@ -135,6 +135,19 @@
      */
     public abstract <U extends BondSerializable> Bonded<U> convert(StructBondType<U> toBondType);
 
+    public <U extends BondSerializable> Bonded<U> cast(StructBondType<U> castBondType) {
+        if (!(getBondType().isSubtypeOf(castBondType))) {
+            final String err = String.format("Bonded<%s> cannot be cast to Bonded<%s>",
+                getBondType().getQualifiedName(), castBondType.getQualifiedName()
+            );
+            throw new ClassCastException(err);
```

Comment:
I don't know. I'm sure `cast()` should throw, because it's supposed to be the same as a real cast. `convert()` returning null seems off to me, but I'm not sure I understand the use case well enough to make proclamations.

If `cast()` fails, you 100% definitely have a broken program. Is the same true for `convert()`?